cache/cache.go

```diff
@@ -191,3 +195,23 @@ func (c *Cache) Path(checksum, uri string) string {
 	base := BasePath(checksum, uri)
 	return filepath.Join(c.root, base)
 }
+
+// UnavailableError returns exit.Unavailable for the exit code.
+type UnavailableError struct {
+	Uri string
+	Err error
+}
+
+// Error returns the error string for the unavailable error
+func (e *UnavailableError) Error() string {
+	msg := fmt.Sprintf("%s is unavailable", e.Uri)
+	if e.Err != nil {
+		return fmt.Sprintf("%s: %v", msg, e.Err)
+	}
+	return msg
+}
+
+// ExitCode returns exit.Unavailable as its exit code
+func (e *UnavailableError) ExitCode() int {
+	return exit.Unavailable
```

Comment:
Let's just use the number directly to avoid adding an extra dep for a single constant?

Reply:
sure